### ISSUE TITLE
Display validation error message

### ISSUE THREAD

**Issue by ecgan** [woocommerce/google-listings-and-ads#1415]
## User story

Currently, we do not have a standard approach and UI design on how to display validation error message.

For example, in Edit Free Listings in the following screenshot, at first it seems like users has entered values for all the fields, but yet the "Save changes" button is disabled, and the users would be left "stucked" with no idea how to proceed or what is wrong. The button is disabled because the users have not entered the shipping times for all the target audience country (the "Add another time" button is the "hint").

![image](https://user-images.githubusercontent.com/417342/162443234-3647f3db-ebf9-43f5-b45e-ff35645761b7.png)

In this issue, we should look into the whole extension and figure out a standard way of displaying validation error message.

Note that in some places, we already have some validation mechanism, e.g. https://github.com/woocommerce/google-listings-and-ads/blob/208bd6a06b25c9893f947a39c1ba0f38f9c391d2/js/src/components/free-listings/configure-product-listings/checkErrors.js, there are some error messages that are set in code but they are never displayed. Devs came up with those error messages on their own, the messages are not from design team or other teams, so we may need to confirm and change those messages accordingly if we were to display them on the UI.

Tagging @j111q @shivanarrthine for your input on the UI design for displaying validation error message.

---

```[tasklist]
### Validatable parts
- [x] Onboarding step 2: Audience, Shipping, and Tax
- [x] Onboarding step 3: Contact information and Pre-launch checklist
- [ ] Onboarding step 4: Ads audience and Budget
- [x] Edit Free Listings: Audience, Shipping, and Tax
- [ ] Attributes (maybe this isn't necessary?)
- [x] Settings: Store address
- [ ] Add/Edit Campaign step 1: Ads audience and Budget
- [x] Add/Edit Campaign step 2: Campaign assets
```

**Comment by mikkamp** [comment]
Just linking a similar issue which highlights the same problem: https://github.com/woocommerce/google-listings-and-ads/issues/634

**Comment by mikkamp** [comment]
This was resolved as part of #1993

When you have forgotten to enter a shipping time or rate for a specific country you will be unable to continue with an error message indicating why:
![image](https://github.com/woocommerce/google-listings-and-ads/assets/11388669/141ae4d4-8f8a-46bc-a432-eaba0a988b4d)
![image](https://github.com/woocommerce/google-listings-and-ads/assets/11388669/89e30b6b-2349-4b16-8ab5-072cc190938e)

Hence I'm closing this issue as completed.
